src/sardana/tango/core/util.py

```diff
@@ -942,18 +947,21 @@ def get_logstash_conf(dev_name):
     if bin_name in ["Pool", "MacroServer"]:
         class_name = bin_name
         dev_name = get_dev_from_class_server(db, class_name, server_name)[0]
-        host, port = get_logstash_conf(dev_name)
+        host, port, cache = get_logstash_conf(dev_name, class_name)
     else:
         dev_name = get_dev_from_class_server(db, "Pool", server_name)[0]
-        host, port = get_logstash_conf(dev_name)
+        host, port, cache = get_logstash_conf(dev_name)
         if host is None:
             dev_name = get_dev_from_class_server(db, "MacroServer",
                                                  server_name)[0]
-            host, port = get_logstash_conf(dev_name)
+            host, port, cache = get_logstash_conf(dev_name)
 
     if host is not None:
         root = Logger.getRootLog()
-        handler = logstash.TCPLogstashHandler(host, port, version=1)
+        handler = AsynchronousLogstashHandler(host, port, database_path=cache)
+        # don't use full path for program_name
+        handler._create_formatter_if_necessary()
```

Comment:
Could you remind me what is it for?

Reply:
This, and the line below is to get only `Pool` and not `/usr/bin/Pool` as `program` keyword.
This worked differently in the synchronous Logstash library.

Reply:
Thanks! Then we need it:)